pkg/api/v1/clients/kube/resource_client.go

```diff
@@ -134,6 +134,18 @@ func (rc *ResourceClient) NewResource() resources.Resource {
 	return resources.Clone(rc.resourceType)
 }
 
+func (rc *ResourceClient) NewInvalidResource() resources.Resource {
+	res, ok := resources.Clone(rc.resourceType).(resources.InputResource)
+	if !ok {
+		return nil
+	}
```

Comment:
suggestion: this part of the code can call `rc.NewResource` and skip the type check
also allows skipping the `nil` check later on

Reply:
Don't I need the type check to access the status functions?

Reply:
Nvm, I see now